### ISSUE TITLE
(1.0 branch) 'new web3' overwrites the previous Contract._eth

### ISSUE THREAD

**Issue by asta331** [ethereum/web3.js#854]
The case:

```
var Web3 = require('./lib/web3.js-1.0/src/index.js');
var tokenContractData = require('./token.js');

var providers = [];

providers.push(new Web3(new Web3.providers.HttpProvider('http://192.168.0.1:8545')));
var contract = new providers[0].eth.Contract(tokenContractData.abi, tokenContractData.address);
console.log(contract._eth._provider);

providers.push(new Web3(new Web3.providers.HttpProvider('http://192.168.0.2:8545')));
console.log(contract._eth._provider);

providers.push(new Web3(new Web3.providers.HttpProvider('http://192.168.0.3:8545')));
console.log(contract._eth._provider);
```

console output:
```
HttpProvider { host: 'http://192.168.0.1:8545', timeout: 0, connected: false }
HttpProvider { host: 'http://192.168.0.2:8545', timeout: 0, connected: false }
HttpProvider { host: 'http://192.168.0.3:8545', timeout: 0, connected: false }
```

**Comment by asta331** [comment]
additional case:
```
var Web3 = require('./lib/web3.js-1.0/src/index.js');
var tokenContractData = require('./token.js');

var providers = [];

providers.push(new Web3(new Web3.providers.HttpProvider('http://192.168.0.1:8545')));
var contract = (new providers[0].eth.Contract(tokenContractData.abi, tokenContractData.address));
contract._eth = providers[0].eth; // this line solved the problem.
console.log(contract._eth._provider);

providers.push(new Web3(new Web3.providers.HttpProvider('http://192.168.0.2:8545')));
console.log(contract._eth._provider);

providers.push(new Web3(new Web3.providers.HttpProvider('http://192.168.0.3:8545')));
console.log(contract._eth._provider);
```

console output:
```
HttpProvider { host: 'http://192.168.0.1:8545', timeout: 0, connected: false }
HttpProvider { host: 'http://192.168.0.1:8545', timeout: 0, connected: false }
HttpProvider { host: 'http://192.168.0.1:8545', timeout: 0, connected: false }
```

**Comment by lionello** [comment]
Yeah, this needs to get fixed. There's a global `currentProvider` which gets overwritten, and this is what `Contract` uses.

**Comment by nivida** [comment]
This should be fixed with the PR #2000

**Comment by nivida** [comment]
This got already fixed. Please update to the latest version of Web3.js.
